orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/model/CIStageDefinition.java

```diff
@@ -41,13 +43,15 @@ public CIStageDefinition(
       @JsonProperty("job") String job,
       @JsonProperty("property") String propertyFile,
       @JsonProperty("buildNumber") Integer buildNumber,
+      @JsonProperty("buildInfo") Map<String, Object> buildInfo,
```

Comment:
I'd like to avoid having to handle ad-hoc `convertValue` operations inside of tasks (as we do for the artifacts in `GetBuildArtifactsTask`), as well as having untyped maps in the stage context.

I think this would be cleaner if we strongly-type the `buildInfo` by adding an inner class below `CIStageDefinition`:
```
  @Getter
  public static class BuildInfo {
    private final ImmutableList<Artifact> artifacts;
    
    public BuildInfo(@JsonProperty("artifacts") List<Artifact> artifacts) {
      this.artifacts =
          Optional.ofNullable(artifacts).map(ImmutableList::copyOf).orElse(ImmutableList.of());
    }
  }
```
and then also in the `CIStageDefinition` constructor just default `buildInfo` to an empty `BuildInfo` if the input is null.

Then we'd only do a single mapping at the beginning of the stage and all the rest of the logic would get type safety and null safety.  (You'd be able to remove the `convertValue` and the null-checks on buildInfo later.)

Reply:
Thanks @ezimanyi! I agree, that is _a lot_ cleaner. Can you please take another look?